spec/unit/command_hierarchy_spec.rb

```diff
@@ -0,0 +1,57 @@
+require '3scale_toolbox/cli'
+
+# `expect(...).not_to matcher.and matcher` is not supported
+RSpec::Matchers.define_negated_matcher :not_raise_error, :raise_error
```

Comment:
Of course. Because every line is this assertion. Raising exception fails the test.

Reply:
When your code can raise `SystemExit`, it is common to add matcher just to make sure `exit` is not raised. If raised, error is catched, other tests are run and everything works as expected. If matcher not added, `SystemExit` makes rspec process to end and many tests are not run.

All tests running
```
ThreeScaleToolbox::CLI.run([name])
```

must include `assert_error` or negative to avoid problems.

Reply:
So you want to test that your code calls or does not call `exit` by rescuing all exceptions? 

Wouldn't it be better just to globally rescue `SystemError` in around block and fail the test?

Then any test that does not assert that it exits would fail. And you don't need negative assertions everywhere - as testing library designers intended.